core/arch/arm/include/kernel/generic_boot.h

```diff
@@ -31,10 +31,12 @@
 
 #if defined(CFG_WITH_ARM_TRUSTED_FW)
 uint32_t cpu_on_handler(uint32_t a0, uint32_t a1);
-uint32_t *generic_boot_init_primary(uint32_t pageable_part);
+uint32_t *generic_boot_init_primary(uint32_t pageable_part, uint32_t unused,
+				    uint32_t fdt);
```

Comment:
Yes I'm mainly concerned with LP64 systems. While unsigned long would be OK, why would `paddr_t` not be? I mean, this *is* a physical address we are expecting, so unless `paddr_t` would not work for some reason, I don't see why we shouldn't use it?